### PR TITLE
[Bazaar] Creating and viewing instant sale items

### PR DESCRIPTION
**Pull request by madsboddum**
Relates to GitHub issue #1227

### CODE REVIEW

File: src/main/java/com/projectswg/holocore/services/gameplay/commodities/BazaarService.kt [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+	}
+
+	private fun getItemDescription(objectById: TangibleObject): String {
+		val instantSaleItem = PswgDatabase.bazaarInstantSales.getInstantSaleItem(objectById.objectId)
+
+		if (instantSaleItem != null) {
+			return instantSaleItem.description
+		}
+
+		return ""
```
Comment (Josh-Larson) [inline src/main/java/com/projectswg/holocore/services/gameplay/commodities/BazaarService.kt:77]: Can be:
```kt
return instantSaleItem?.description ?: ""
```

File: src/main/java/com/projectswg/holocore/services/gameplay/commodities/BazaarService.kt [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+	private fun handleMySalesWindow(packet: AuctionQueryHeadersMessage, player: Player) {
+		val auctionQueryHeadersResponseMessage = AuctionQueryHeadersResponseMessage()
+		auctionQueryHeadersResponseMessage.updateCounter = packet.updateCounter
+		auctionQueryHeadersResponseMessage.windowType = packet.windowType
+		val instantSaleItems = PswgDatabase.bazaarInstantSales.getMyInstantSaleItems(player.creatureObject.objectId)
+		val now = LocalDateTime.now()
+
+		for (instantSaleItem in instantSaleItems) {
+			val expiresInSeconds = now.until(instantSaleItem.expiresAt, SECONDS).toInt()
+			if (expiresInSeconds >= 60) {    // Client crashes if this ever goes below 60
```
Comment (Josh-Larson) [inline src/main/java/com/projectswg/holocore/services/gameplay/commodities/BazaarService.kt:161]:
```kt
if (expiresInSeconds < 60)
    continue
```

File: src/main/java/com/projectswg/holocore/services/gameplay/commodities/BazaarService.kt [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+	private fun handleAllAuctionsWindow(packet: AuctionQueryHeadersMessage, player: Player) {
+		val auctionQueryHeadersResponseMessage = AuctionQueryHeadersResponseMessage()
+		auctionQueryHeadersResponseMessage.updateCounter = packet.updateCounter
+		auctionQueryHeadersResponseMessage.windowType = packet.windowType
+		val instantSaleItems = PswgDatabase.bazaarInstantSales.getInstantSaleItems()
+		val now = LocalDateTime.now()
+
+		for (instantSaleItem in instantSaleItems) {
+			val expiresInSeconds = now.until(instantSaleItem.expiresAt, SECONDS).toInt()
+			if (expiresInSeconds >= 60) {    // Client crashes if this ever goes below 60
```
Comment (Josh-Larson) [inline src/main/java/com/projectswg/holocore/services/gameplay/commodities/BazaarService.kt:205]:
```kt
if (expiresInSeconds < 60)
    continue
```

Comment (Josh-Larson) [reply src/main/java/com/projectswg/holocore/services/gameplay/commodities/BazaarService.kt:205]: There appears to be a fair amount of duplication between these two functions, can they be combined into a separate function?

Comment (madsboddum) [reply src/main/java/com/projectswg/holocore/services/gameplay/commodities/BazaarService.kt:205]: The only difference is where the instant sale items come from on the bazaar item database class. Definitely makes to combine them.

**Pull request merged**